### DISCUSSION TITLE
How to set interactor_settings in VtkLocalView to map mouse buttons and thereby achieve the functionality of vtkInteractorStyleTrackballActor?

### DISCUSSION THREAD

**Discussion by pengwuke** [Kitware/trame#405] [General]
![image](https://github.com/Kitware/trame/assets/30679728/c82a09d2-2efb-4d25-9a7e-ddd179957087)
https://github.com/pyvista/pyvista/issues/5432
This is implemented using PyVista, but I'm also encountering interactivity issues within VtkLocalView.
![image](https://github.com/Kitware/trame/assets/30679728/1a01ca25-9af6-4652-8a08-9ae29c716018)

**Comment by jourdain** [comment]
While it is possible to tune the interaction mapping the InteractorStyleTrackballActor is not exposed but could be added.

From the trame perspective, you can adjust the interaction mapping by setting the [`interactor_settings=...`](https://github.com/Kitware/vue-vtk-js/blob/master/src/core/VtkSyncView.js#L24-L62) property. 

To add the InteractorStyleTrackballActor as a new action like __RotateActor__, we will need at least 2 things.
1. Update vtk.js manipulator to expose such InteractorStyleTrackballActor behavior as a manipulator inside [here](https://github.com/Kitware/vtk-js/tree/master/Sources/Interaction/Manipulators)
2. Register that manipulator [here](https://github.com/Kitware/vue-vtk-js/blob/master/src/core/localview.js#L40-L49)
3. Update the hierarchy of libraries to trigger a new set of releases (vtk.js, vue-vtk-js, trame-vtk)
4. Update PyVista to configure that interaction mapping based on its settings automatically

HTH

**Reply by pengwuke** [reply]
Thank you very much for your patient explanation
